src/ManifestStrategy/GlossaryManifestStrategy.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace SprykerSdk\Integrator\ManifestStrategy;
+
+use SprykerSdk\Integrator\Dependency\Console\InputOutputInterface;
+
+class GlossaryManifestStrategy extends AbstractManifestStrategy
+{
+    /**
+     * @var int
+     */
+    protected const GLOSSARY_LINE_PARTS_COUNT = 3;
+
+    /**
+     * @return string
+     */
+    public function getType(): string
+    {
+        return 'glossary-key';
+    }
+
+    /**
+     * @param array<string, array<string, string>> $manifest
+     * @param string $moduleName
+     * @param \SprykerSdk\Integrator\Dependency\Console\InputOutputInterface $inputOutput
+     * @param bool $isDry
+     *
+     * @return bool
+     */
+    public function apply(array $manifest, string $moduleName, InputOutputInterface $inputOutput, bool $isDry): bool
+    {
+        $glossaryFilePath = $this->config->getGlossaryFilePath();
+        if (!file_exists($glossaryFilePath)) {
+            $inputOutput->writeln(sprintf(
+                'File `%s` does not exist. Please check file path or customize using `IntegratorConfig::getGlossaryFilePath()`.',
+                $glossaryFilePath,
+            ), InputOutputInterface::DEBUG);
+
+            return false;
+        }
+
+        $existingGlossaryFileLines = $this->getGlossaryExistingFileLines();
+        $glossaryFileLinesForAdd = $this->getGlossaryFileLinesForAdd(
+            $manifest,
+            $existingGlossaryFileLines,
+        );
+        $resultGlossaryFileLines = array_merge($existingGlossaryFileLines, $glossaryFileLinesForAdd);
+        if (!$isDry) {
+            file_put_contents($this->config->getGlossaryFilePath(), implode(PHP_EOL, $resultGlossaryFileLines));
+        }
+        $inputOutput->writeln(sprintf(
+            'Glossary file `%s` successfully updated',
+            $glossaryFilePath,
+        ), InputOutputInterface::DEBUG);
+
+        return true;
+    }
+
+    /**
+     * @param array<string, array<string, string>> $manifest
+     * @param array<array-key, string> $existingGlossaryFileLines
+     *
+     * @return array<array-key, string>
+     */
+    protected function getGlossaryFileLinesForAdd(
+        array $manifest,
+        array $existingGlossaryFileLines
+    ): array {
+        $indexGlossaryLinesByGlossaryKeysAndLanguages = $this->indexGlossaryLinesByGlossaryKeysAndLanguages(
```

Comment:
minor, but the name of the variable should be noun. We can read indexGlossaryLinesByGlossaryKeysAndLanguages also as a noun, but it has a little bit different meaning.

IMO it should be indexedGlossaryLinesByGlossaryKeysAndLanguages